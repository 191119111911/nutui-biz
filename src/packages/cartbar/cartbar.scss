@import '../cartbaricon/cartbaricon.scss';
@import '../cartbarbutton/cartbarbutton.scss';
@import '../../styles/base.scss';

.nut-cart-bar {
    .nut-cart-bar-inner {
        display: flex;

        &.has-capsule-buttons {
            > article {
                margin-right: 0;

                .nut-button--round {
                    border-radius: 0;
                }
    
                &:first-of-type {
                    .nut-button--round {
                        border-top-left-radius: 25px;
                        border-bottom-left-radius: 25px;
                    }
                }
            
                &:last-of-type {
                    .nut-button--round {
                        border-top-right-radius: 25px;
                        border-bottom-right-radius: 25px;
                    }
                }
            }
        }
    }
}