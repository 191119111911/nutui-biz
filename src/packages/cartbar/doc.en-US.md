#  CartBar 购物车栏

### Intro

Commonly found at the bottom of product detail pages, including a set of icons (contact customer service, shopping cart, etc.) and a set of buttons (add cart, buy it now).

### Install

```javascript
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';
```

## Demo

### Basic Usage

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon text="shop" iconProps={{name: 'shop'}} />
      <CartBarIcon text="shopping cart" iconProps={{name: 'cart'}} />
      <CartBarButton text="add cart" buttonProps={{type: 'danger'}} />
      <CartBarButton text="buy it now" buttonProps={{type: 'warning'}} />
    </CartBar>
  );
};
export default App;

// css
[class*="safe-area-bottom"] {
    position: relative;
}

```

:::

### With Badge

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon text="shop" iconProps={{name: 'shop'}} badgeProps={{value: '10'}}  />
      <CartBarIcon text="shopping cart" iconProps={{name: 'cart'}} badgeProps={{dot: true}} />
      <CartBarButton text="add cart" buttonProps={{type: 'danger'}} />
      <CartBarButton text="buy it now" buttonProps={{type: 'warning'}} />
    </CartBar>
  );
};
export default App;

// css
[class*="safe-area-bottom"] {
    position: relative;
}

```

:::

### Custom Icon Color

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon text="shop" iconProps={{name: 'shop', color: 'red'}} />
      <CartBarIcon text="shopping cart" iconProps={{name: 'cart'}} />
      <CartBarButton text="add cart" buttonProps={{type: 'danger'}} />
      <CartBarButton text="buy it now" buttonProps={{type: 'warning'}} />
    </CartBar>
  );
};
export default App;

// css
[class*="safe-area-bottom"] {
    position: relative;
}

```

:::

### Icon Without Text

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon iconProps={{name: 'shop'}} />
      <CartBarIcon iconProps={{name: 'cart'}} />
      <CartBarButton text="add cart" buttonProps={{type: 'danger'}} />
      <CartBarButton text="buy it now" buttonProps={{type: 'warning'}} />
    </CartBar>
  );
};
export default App;

// css
[class*="safe-area-bottom"] {
    position: relative;
}

```

:::

### Capsule Button

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar hasCapsuleButtons>
      <CartBarIcon text="shop" iconProps={{name: 'shop'}} />
      <CartBarIcon text="shopping cart" iconProps={{name: 'cart'}} />
      <CartBarButton text="add cart" buttonProps={{type: 'danger'}} />
      <CartBarButton text="buy it now" buttonProps={{type: 'warning'}} />
    </CartBar>
  );
};
export default App;

// css
[class*="safe-area-bottom"] {
    position: relative;
}

```

:::

### Custom Top Content

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  const customTopHtml = () => {
    return <div style={{display: 'flex', height: '30px', alignItems: 'center', justifyContent: 'center', color: 'red'}}>Custom top content.</div>
  }

  return (
    <CartBar top={customTopHtml()}>
      <CartBarIcon text="shop" iconProps={{name: 'shop'}} />
      <CartBarIcon text="shopping cart" iconProps={{name: 'cart'}} />
      <CartBarButton text="add cart" buttonProps={{type: 'danger'}} />
      <CartBarButton text="buy it now" buttonProps={{type: 'warning'}} />
    </CartBar>
  );
};
export default App;

// css
[class*="safe-area-bottom"] {
    position: relative;
}

```

:::




## API


### CartBar Props


| Attribute    | Description                                       | Type    | Default    |
|---------|--------------------------------------------|---------|-----------|
| hasCapsuleButtons   | Whether to include capsule buttons                                 | boolean  | `false`          |
| safeAreaInsetBottom   | Whether to enable bottom safe area adaptation                                 | boolean  | `true`          |
| placeholder   | Whether to generate a placeholder element when fixed                                 | boolean  | `false`          |
| top | Custom top content |ReactNode  | -          |


### CartBarIcon Props


| Attribute    | Description                                       | Type    | Default    |
|---------|--------------------------------------------|---------|-----------|
| text   | Icon text                                 | ReactNode  | -         |
| iconProps   | Icon props                                 | [IconProps](https://nutui.jd.com/h5/react/1x/#/zh-CN/component/icon)  | -         |
| badgeProps   | Badge props                                 | [BadgeProps](https://nutui.jd.com/h5/react/1x/#/zh-CN/component/badge)  | -         |


### CartBarIcon Events
| Attribute | Description | Arguments |
|----- | ----- | -----  |
| onClick | Click event |  - |



### CartBarButton Props


| Attribute    | Description                                       | Type    | Default    |
|---------|--------------------------------------------|---------|-----------|
| text   | Button text                                 | ReactNode  | -         |
| buttonProps   | Button props                                 | [ButtonProps](https://nutui.jd.com/h5/react/1x/#/zh-CN/component/button)  | -         |


### CartBarButton Events
| Attribute | Description | Arguments |
|----- | ----- | -----  |
| onClick | Click event |  - |