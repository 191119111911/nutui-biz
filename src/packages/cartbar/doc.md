#  CartBar 购物车栏

### 介绍

常见于商详页底部，包括一组图标（联系客服、购物车等）和一组按钮（加车、立即购买）。

### 安装

```javascript
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';
```

## 代码演示

### 基本用法

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon text="店铺" icon="shop" />
      <CartBarIcon text="购物车" icon="cart" />
      <CartBarButton text="加入购物车" type="danger" />
      <CartBarButton text="立即购买" type="warning" />
    </CartBar>
  );
};
export default App;
```

:::

### 带有徽标

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon text="店铺" icon="shop" badge="10" />
      <CartBarIcon text="购物车" icon="cart" dot />
      <CartBarButton text="加入购物车" type="danger" />
      <CartBarButton text="立即购买" type="warning" />
    </CartBar>
  );
};
export default App;
```

:::

### 自定义图标颜色

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon text="店铺" color="red" icon="shop" />
      <CartBarIcon text="购物车" icon="cart" />
      <CartBarButton text="加入购物车" type="danger" />
      <CartBarButton text="立即购买" type="warning" />
    </CartBar>
  );
};
export default App;
```

:::

### 图标无文本

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar>
      <CartBarIcon icon="shop" />
      <CartBarIcon icon="cart" />
      <CartBarButton text="加入购物车" type="danger" />
      <CartBarButton text="立即购买" type="warning" />
    </CartBar>
  );
};
export default App;
```

:::

### 胶囊型按钮

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  return (
    <CartBar hasCapsuleButtons>
      <CartBarIcon text="店铺" icon="shop" />
      <CartBarIcon text="购物车" icon="cart" />
      <CartBarButton text="加入购物车" type="danger" />
      <CartBarButton text="立即购买" type="warning" />
    </CartBar>
  );
};
export default App;
```

:::

### 顶部自定义内容

:::demo

```ts
import  React from 'react';
import { CartBar, CartBarIcon, CartBarButton } from '@nutui/nutui-biz';

const App = () => {
  const customTopHtml = () => {
    return <div style={{display: 'flex', height: '30px', alignItems: 'center', justifyContent: 'center', color: 'red'}}>我是自定义内容！</div>
  }

  return (
    <CartBar top={customTopHtml()}>
      <CartBarIcon text="店铺" icon="shop" />
      <CartBarIcon text="购物车" icon="cart" />
      <CartBarButton text="加入购物车" type="danger" />
      <CartBarButton text="立即购买" type="warning" />
    </CartBar>
  );
};
export default App;
```

:::




## API


### CartBar Props


| 字段    | 说明                                       | 类型    | 默认值    |
|---------|--------------------------------------------|---------|-----------|
| hasCapsuleButtons   | 是否包含胶囊型按钮                                 | Boolean  | `false`          |
| safeAreaInsetBottom   | 是否开启iphone系列全面屏底部安全区适配                                 | Boolean  | `true`          |
| placeholder   | 固定在底部时，是否在标签位置生成一个等高的占位元素                                 | Boolean  | `false`          |
| top | 自定义顶部内容 |React.ReactNode  | -          |


### CartBarIcon Props


| 字段    | 说明                                       | 类型    | 默认值    |
|---------|--------------------------------------------|---------|-----------|
| text   | 图标文字                                 | String  | -         |
| icon   | 图标                                 | String  | -         |
| color   | 图标颜色                                 | String  | `#323233`         |
| dot   | 是否显示图标右上角小红点                                 | Boolean  | `false`         |
| badge   | 图标右上角徽标的内容	                                 | Number \| String  | -         |


## CartBarIcon Events
| 字段 | 说明 | 回调参数 |
|----- | ----- | -----  |
| onClick | 点击事件 |  event: Event |



### CartBarButton Props


| 字段    | 说明                                       | 类型    | 默认值    |
|---------|--------------------------------------------|---------|-----------|
| text   | 按钮文字                                 | String  | -         |
| disabled   | 是否禁用按钮                                 | Boolean  | `false`         |
| type   | 按钮类型，可选值为 `primary` `info` `warning` `danger`                                 | String  | `default`         |
| loading   | 是否显示为加载状态                                 | Boolean  | `false`         |
| color   | 图标颜色                                 | String  | -         |


## CartBarButton Events
| 字段 | 说明 | 回调参数 |
|----- | ----- | -----  |
| onClick | 点击事件 |  event: Event |