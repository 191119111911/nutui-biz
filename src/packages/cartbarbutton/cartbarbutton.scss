.nb-cart-bar-button {
    margin-right: 5px;

    &:last-child {
        margin-right: 0;
    }
}