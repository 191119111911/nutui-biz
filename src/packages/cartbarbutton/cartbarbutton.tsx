import React, {
  FunctionComponent,
  ReactNode
} from 'react'
import classNames from 'classnames';
import {cn2} from '@/utils/bem'
import {Button, ButtonProps} from '@nutui/nutui-react'

import { IComponent } from '@/utils/typings'

export interface CartBarButtonProps extends IComponent {
  text: ReactNode
  buttonProps: Partial<ButtonProps>
  onClick: () => void
}

const defaultProps = {
  onClick: () => {}
} as CartBarButtonProps

export const CartBarButton: FunctionComponent<
  Partial<CartBarButtonProps>
> = (props) => {
  const {
    className,
    style,
    text,
    onClick,
    buttonProps,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }

  const b = cn2('cart-bar-button')

  const handleClick = () => {
    onClick()
  }

  return (
    <p className={classNames(b(),className)} style={style} {...rest} onClick={handleClick}>
      <Button 
        {...buttonProps}
      >
        {text}
      </Button>
    </p>
  )
}

CartBarButton.defaultProps = defaultProps
CartBarButton.displayName = 'NutCartBarButton'
