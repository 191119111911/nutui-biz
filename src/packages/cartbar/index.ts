import { CartBar } from './cartbar'

export default CartBar
