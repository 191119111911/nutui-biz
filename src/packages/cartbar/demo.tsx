import React from 'react'
import { CartBar } from './cartbar'
import { CartBarIcon } from '../cartbaricon/cartbaricon'
import { CartBarButton } from '../cartbarbutton/cartbarbutton'
import { useTranslate } from '../../sites/assets/locale'

interface tarnslatedOption {
  basic: string
  title1: string
  title2: string
  title3: string
  title4: string
  title5: string
  shop: string
  shoppingCart: string
  addCart: string
  buyItNow: string
}

const CartBarDemo = () => {
  const [translated] = useTranslate<tarnslatedOption>({
    'zh-CN': {
      basic: '基本用法',
      title1: '带有徽标',
      title2: '自定义图标颜色',
      title3: '图标无文本',
      title4: '胶囊型按钮',
      title5: '顶部自定义内容',
      shop: '店铺',
      shoppingCart: '购物车',
      addCart: '加入购物车',
      buyItNow: '立即购买'
    },
    'en-US': {
      basic: 'Basic Usage',
      title1: 'With Badge',
      title2: 'Custom Icon Color',
      title3: 'Icon Without Text',
      title4: 'Capsule Button',
      title5: 'Custom Top Content',
      shop: 'shop',
      shoppingCart: 'shoppingCart',
      addCart: 'add cart',
      buyItNow: 'buy it now'
    }
  });
  const customTopHtml = () => {
    return <div style={{display: 'flex', height: '30px', alignItems: 'center', justifyContent: 'center', color: 'red'}}>{translated.title5}</div>
  }
  
  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <CartBar>
          <CartBarIcon text={translated.shop} icon="shop" />
          <CartBarIcon text={translated.shoppingCart} icon="cart" />
          <CartBarButton text={translated.addCart} buttonProps={{type: 'danger'}} />
          <CartBarButton text={translated.buyItNow} buttonProps={{type: 'warning'}} />
        </CartBar>
        <h2>{translated.title1}</h2>
        <CartBar>
          <CartBarIcon text={translated.shop} icon="shop" badge="10" />
          <CartBarIcon text={translated.shoppingCart} icon="cart" dot />
          <CartBarButton text={translated.addCart} buttonProps={{type: 'danger'}} />
          <CartBarButton text={translated.buyItNow} buttonProps={{type: 'warning'}} />
        </CartBar>
        <h2>{translated.title2}</h2>
        <CartBar>
          <CartBarIcon text={translated.shop} color="red" icon="shop" />
          <CartBarIcon text={translated.shoppingCart} icon="cart" />
          <CartBarButton text={translated.addCart} buttonProps={{type: 'danger'}} />
          <CartBarButton text={translated.buyItNow} buttonProps={{type: 'warning'}} />
        </CartBar>
        <h2>{translated.title3}</h2>
        <CartBar>
          <CartBarIcon icon="shop" />
          <CartBarIcon icon="cart" />
          <CartBarButton text={translated.addCart} buttonProps={{type: 'danger'}} />
          <CartBarButton text={translated.buyItNow} buttonProps={{type: 'warning'}} />
        </CartBar>
        <h2>{translated.title4}</h2>
        <CartBar hasCapsuleButtons>
          <CartBarIcon text={translated.shop} icon="shop" />
          <CartBarIcon text={translated.shoppingCart} icon="cart" />
          <CartBarButton text={translated.addCart} buttonProps={{type: 'danger'}} />
          <CartBarButton text={translated.buyItNow} buttonProps={{type: 'warning'}} />
        </CartBar>
        <h2>{translated.title5}</h2>
        <CartBar top={customTopHtml()}>
          <CartBarIcon text={translated.shop} icon="shop" />
          <CartBarIcon text={translated.shoppingCart} icon="cart" />
          <CartBarButton text={translated.addCart} buttonProps={{type: 'danger'}} />
          <CartBarButton text={translated.buyItNow} buttonProps={{type: 'warning'}} />
        </CartBar>
      </div>
    </>
  )
}

export default CartBarDemo
