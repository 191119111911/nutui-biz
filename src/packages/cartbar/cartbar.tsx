import React, {
  FunctionComponent,
} from 'react'
import { useConfig } from '@/packages/configprovider'

import { IComponent } from '@/utils/typings'

export interface CartBarProps extends IComponent {
}

const defaultProps = {
} as CartBarProps

export const CartBar: FunctionComponent<
  Partial<CartBarProps> & Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'>
> = (props) => {
  const { locale } = useConfig()
  const {
  } = {
    ...defaultProps,
    ...props,
  }

  return (
    <div>
    </div>
  )
}

CartBar.defaultProps = defaultProps
CartBar.displayName = 'NutCartBar'
