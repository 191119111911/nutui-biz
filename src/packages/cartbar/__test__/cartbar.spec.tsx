import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { CartBar } from '../cartbar'
import { CartBarIcon } from '../../cartbaricon/cartbaricon'
import { CartBarButton } from '../../cartbarbutton/cartbarbutton'

test('test base', async () => {
  const { container } = render(
    <CartBar>
      <CartBarIcon text="店铺" icon="shop" />
      <CartBarIcon text="购物车" icon="cart" />
      <CartBarButton text="加入购物车" buttonProps={{type: "danger"}} />
      <CartBarButton text="立即购买" buttonProps={{type: "warning"}} />
    </CartBar>
  )
  const cartBar = container.querySelector('.nb-cart-bar')
  expect(cartBar).toHaveClass('nut-biz-safe-area-bottom')
  expect(cartBar).toBeTruthy()
  const cartBarIcons = container.querySelectorAll('.nb-cart-bar-icon')
  const cartBarButtons = container.querySelectorAll('.nb-cart-bar-button')
  expect(cartBarIcons.length).toBe(2)
  expect(cartBarButtons.length).toBe(2)
})

test('test icon badge and dot', async () => {
  const { container } = render(
    <CartBar>
      <CartBarIcon text="店铺" icon="shop" badge="10" />
      <CartBarIcon text="购物车" icon="cart" dot />
    </CartBar>
  )

  const cartBarIcons = container.querySelectorAll('.nb-cart-bar-icon')
  
  expect(cartBarIcons[0].querySelector('.nut-badge__content')?.innerHTML).toBe('10')
  expect(cartBarIcons[1].querySelector('.nut-badge__content')).toHaveClass('is-dot')
})

test('test icon color', async () => {
  const { container } = render(
    <CartBar>
      <CartBarIcon text="店铺" icon="shop" color='red' />
    </CartBar>
  )

  const iconfont = container.querySelector('.nb-cart-bar-icon .nutui-iconfont')
  
  expect(iconfont).toHaveAttribute('style', 'color: red;')
})

test('test no text icon', async () => {
  const { container } = render(
    <CartBar>
      <CartBarIcon icon="shop" />
    </CartBar>
  )

  const txt = container.querySelector('.txt')
  expect(txt).not.toBeTruthy()
})

test('test hasCapsuleButtons', async () => {
  const { container } = render(
    <CartBar hasCapsuleButtons>
      <CartBarButton text="加入购物车"  buttonProps={{type: "danger"}} />
      <CartBarButton text="立即购买"  buttonProps={{type: "warning"}} />
    </CartBar>
  )

  const cartBarInner = container.querySelector('.nb-cart-bar__inner')
  expect(cartBarInner).toHaveClass('has-capsule-buttons')
})

test('test hasCapsuleButtons', async () => {
  const { container } = render(
    <CartBar top={<div>我是自定义内容！</div>}>
      <CartBarIcon text="店铺" icon="shop" />
      <CartBarIcon text="购物车" icon="cart" />
      <CartBarButton text="加入购物车"  buttonProps={{type: "danger"}} />
      <CartBarButton text="立即购买"  buttonProps={{type: "warning"}} />
    </CartBar>
  )

  const cartBar = container.querySelector('.nb-cart-bar')
  expect(cartBar).toHaveTextContent('我是自定义内容！')
})

test('test placeholder', async () => {
  const { container } = render(
    <CartBar placeholder>
      <CartBarIcon text="店铺" icon="shop" />
    </CartBar>
  )

  expect(container.querySelector('.nut-cart-bar--placeholder')?.innerHTML).toMatchSnapshot()
})

test('test click', async () => {
  const onClick1 = jest.fn()
  const onClick2 = jest.fn()
  const { container } = render(
    <CartBar>
      <CartBarIcon text="店铺" icon="shop" onClick={onClick1} />
      <CartBarButton text="加入购物车"  buttonProps={{type: "danger"}} onClick={onClick2} />
    </CartBar>
  )

  const cartBarIcon = container.querySelector('.nb-cart-bar-icon') as HTMLElement
  const cartBarButton = container.querySelector('.nb-cart-bar-button') as HTMLElement
  fireEvent.click(cartBarIcon)
  fireEvent.click(cartBarButton)
  expect(onClick1).toBeCalled()
  expect(onClick2).toBeCalled()
})