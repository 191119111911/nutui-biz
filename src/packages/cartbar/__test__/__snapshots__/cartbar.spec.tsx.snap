// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test placeholder 1`] = `undefined`;
